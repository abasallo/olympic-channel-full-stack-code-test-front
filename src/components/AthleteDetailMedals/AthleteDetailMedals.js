import React from 'react'

import PropTypes from 'prop-types'

import TableCell from '@material-ui/core/TableCell'
import TableContainer from '@material-ui/core/TableContainer'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import TableBody from '@material-ui/core/TableBody'

import { StyledTable, StyledPaper } from './AthleteDetailMedals.styled.components'

const AthleteDetailMedals = (props) => (
  <div>
    <h1>Medals</h1>
    <TableContainer component={StyledPaper}>
      <StyledTable>
        <TableHead>
          <TableRow>
            <TableCell align="center">Year</TableCell>
            <TableCell align="center">Game</TableCell>
            <TableCell align="center">Gold</TableCell>
            <TableCell align="center">Silver</TableCell>
            <TableCell align="center">Bronze</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {props.athleteResults.map((result) => (
            <TableRow key={result.id}>
              <TableCell align="center">{result.game.year}</TableCell>
              <TableCell align="center">{result.game.city}</TableCell>
              <TableCell align="center">{result.gold}</TableCell>
              <TableCell align="center">{result.silver}</TableCell>
              <TableCell align="center">{result.bronze}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </StyledTable>
    </TableContainer>
  </div>
)

AthleteDetailMedals.propTypes = { athleteResults: PropTypes.array }

export default AthleteDetailMedals
