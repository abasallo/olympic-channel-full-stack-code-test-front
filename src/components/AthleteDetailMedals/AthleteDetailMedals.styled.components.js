import styled from 'styled-components'

import Table from '@material-ui/core/Table'
import Paper from '@material-ui/core/Paper'

export const StyledTable = styled(Table)`
  text-align: center;
`

export const StyledPaper = styled(Paper)`
  width: 50%;
  display: block;
  margin-left: auto;
  margin-right: auto;
`
