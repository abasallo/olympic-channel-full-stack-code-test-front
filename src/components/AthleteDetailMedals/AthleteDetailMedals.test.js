import React from 'react'

import AthleteDetailMedals from './AthleteDetailMedals'

const athleteResults = [
  { id: '1-4', gold: 1, silver: 1, bronze: 1, game: { id: 4, city: 'Pyeongchang', year: 2018 } },
  { id: '1-5', gold: 0, silver: 1, bronze: 2, game: { id: 5, city: 'Sochi', year: 2014 } },
  { id: '1-6', gold: 0, silver: 0, bronze: 1, game: { id: 6, city: 'Vancouver', year: 2010 } },
  { id: '1-7', gold: 0, silver: 0, bronze: 1, game: { id: 7, city: 'Turin', year: 2006 } }
]

test('Renders properly', () => expect(<AthleteDetailMedals athleteResults={athleteResults} />).toMatchSnapshot())
