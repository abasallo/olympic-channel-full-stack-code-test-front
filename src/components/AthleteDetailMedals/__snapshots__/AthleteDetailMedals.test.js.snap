// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<AthleteDetailMedals
  athleteResults={
    Array [
      Object {
        "bronze": 1,
        "game": Object {
          "city": "Pyeongchang",
          "id": 4,
          "year": 2018,
        },
        "gold": 1,
        "id": "1-4",
        "silver": 1,
      },
      Object {
        "bronze": 2,
        "game": Object {
          "city": "Sochi",
          "id": 5,
          "year": 2014,
        },
        "gold": 0,
        "id": "1-5",
        "silver": 1,
      },
      Object {
        "bronze": 1,
        "game": Object {
          "city": "Vancouver",
          "id": 6,
          "year": 2010,
        },
        "gold": 0,
        "id": "1-6",
        "silver": 0,
      },
      Object {
        "bronze": 1,
        "game": Object {
          "city": "Turin",
          "id": 7,
          "year": 2006,
        },
        "gold": 0,
        "id": "1-7",
        "silver": 0,
      },
    ]
  }
/>
`;
